### PR TITLE
Use concise date format when plotting

### PR DESCRIPTION
**Pull request by Illviljan**
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

```python
import matplotlib.pyplot as plt
import xarray as xr

airtemps = xr.tutorial.open_dataset("air_temperature")
air = airtemps.air - 273.15
air1d = air.isel(lat=10, lon=10)

plt.figure()
air1d.plot()
```
Before:
image

After:
image

### CODE REVIEW

**Comment by dcherian** [comment]
Looks great,

@jklymak is there a reason this isn't the default in matplotlib yet?

**Comment by jklymak** [comment]
@dcherian Largely back compatibility concerns.   Matplotlib doesn't like suddenly changing the appearance of plots if possible.  I think its on the list for a style-break release.

File: xarray/tests/test_plot.py [hunk 1]
```diff
@@ -1888,6 +1898,21 @@ def test_interval_breaks_logspace(self) -> None:
 class TestImshow(Common2dMixin, PlotTestCase):
     plotfunc = staticmethod(xplt.imshow)
 
+    @pytest.mark.xfail(
+        reason="Failing, should work. remove when works, already in Common2dMixin"
```
Comment (headtr1ck) [inline xarray/tests/test_plot.py:1902]: Is this PR already ready? If so, why are these failing? Can you add a better reason?

Comment (Illviljan) [reply xarray/tests/test_plot.py:1902]: They have always failed, the old inherited tests only did `a.plot(x="t")`. 

Try on main:
```python
import pandas as pd
import xarray as xr
import numpy as np

time = pd.date_range("2000-01-01", "2000-01-10")
a = DataArray(np.random.randn(2, len(time)), [("xx", [1, 2]), ("t", time)])
a.plot.imshow(x="t") # Fails inside matplotlib :(

a.plot.contourf(x="t") # Works! :)
```

Comment (Illviljan) [reply xarray/tests/test_plot.py]: Clarified the reason a little. Is it any better?

Comment (headtr1ck) [reply xarray/tests/test_plot.py]: thanks! Didn't realize that they were failing before, sry.

**Review by headtr1ck — APPROVED** [review]
LGTM, thanks!

**Pull request merged**